### ISSUE TITLE
Error with ExAws.request!/2 for Elixir 1.6.6 & Erlang 21.0

### ISSUE THREAD

**Issue by supercodepoet** [ex-aws/ex_aws#567]
### Environment

* Elixir 1.6.6
* Erlang/OTP 21
* ex_aws 2.0.2 (Hex package) (mix)
* ex_aws_s3 2.0.0 (Hex package) (mix)
* hackney 1.12.1 (Hex package) (rebar3)

### Current behavior

Getting the following error using `ExAws.request!/2`:

** (FunctionClauseError) no function clause matching in :raw_file_io_raw.open_layer/3
        (kernel) raw_file_io_raw.erl:24: :raw_file_io_raw.open_layer("priv/static/css/app.css", [:read, :read_ahead, {:read_ahead, 65536}, :binary, :binary], [:raw, :raw])
        (elixir) lib/file/stream.ex:78: anonymous fn/2 in Enumerable.File.Stream.reduce/3
        (elixir) lib/stream.ex:1346: anonymous fn/5 in Stream.resource/3
        (elixir) lib/stream.ex:1536: Enumerable.Stream.do_each/4
        (elixir) lib/task/supervised.ex:295: Task.Supervised.stream_reduce/7
        (elixir) lib/enum.ex:1919: Enum.reverse/1
        (elixir) lib/enum.ex:2588: Enum.to_list/1
        lib/ex_aws/s3/upload.ex:89: ExAws.Operation.ExAws.S3.Upload.perform/2
        (ex_aws) lib/ex_aws.ex:61: ExAws.request!/2

**Comment by benwilson512** [comment]
Please provide the full code that was run to produce this error.

**Comment by supercodepoet** [comment]
```elixir
def publish_to_s3(:file, file, bucket, relative_to) when is_binary(file) do
    response =
      file
      |> Upload.stream_file()
      |> S3.upload(
        bucket,
        Path.relative_to(file, relative_to),
        acl: :public_read,
        cache_control: cache_control(file),
        content_type: MIME.from_path(file)
      )
      |> ExAws.request!()

    Mix.Shell.IO.info("...published #{file}")

    {:ok, response}
  end
```

**Comment by benwilson512** [comment]
This may be a bug in Elixir: https://github.com/elixir-lang/elixir/issues/7771

**Comment by benwilson512** [comment]
Fixed by https://github.com/ex-aws/ex_aws_s3/pull/22

**Comment by wmnnd** [comment]
@benwilson512  Would you consider releasing `1.1.5` with this patch included as `1.1.6`?

**Comment by benwilson512** [comment]
@wmnnd ExAws v2 has been out for 8 or 9 months at this point, and is plug and play compatible with V1 if you install the requisite other packages. I do not intend on supporting V1 still sorry.
